.github/workflows/deploy-docs.yml

```diff
@@ -0,0 +1,57 @@
+name: Build and Deploy Docs
+on:
+  push:
+    branches:
+      - main
+jobs:
+  build-and-deploy:
+    name: Build and Deploy
+    runs-on: ubuntu-20.04
+    env:
+      MIKE_VERSION: v2
+      MIKE_ALIAS: latest
+      DOCKER_IMAGE: ui5-tooling/mkdocs-material
+      GIT_COMMITTER_NAME: "OpenUI5 Bot"
+      GIT_COMMITTER_EMAIL: "openui5@sap.com"
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Install npm dependencies
+      run: npm ci
+
+    - name: GitHub Configuration
+      run: git fetch origin gh-pages --depth=1
+
+    - name: Build docs with Mike
+      run: ./scripts/buildDocs.sh
+
+    - name: Publish docs
+      run: docker run --rm -v $(pwd):/docs --entrypoint mike --env GIT_COMMITTER_NAME="${GIT_COMMITTER_NAME}" --env GIT_COMMITTER_EMAIL="${GIT_COMMITTER_EMAIL}" $DOCKER_IMAGE set-default $MIKE_VERSION --push
+
+    - name: Set /site ownership to current user
+      run: sudo chown -R $(id -u):$(id -g) ./site
+    - name: Build JSDoc
+      run: npm run jsdoc-generate
+    - name: Build Schema
+      run: npm run schema-generate
+
+    - name: Checkout gh-pages
+      uses: actions/checkout@v3
+      with:
+        ref: gh-pages
+        path: gh-pages
+    - name: Copy the API & JSON Schema to gh-pages
+      run: |
+        cp -R ./site/schema ./gh-pages/ 2>/dev/null || :
```

Comment:
I was wondering how to properly handle those.
It copies the results/outputs from the previous steps `Build JSDoc` & `Build Schema` to the correct locations. 
The problem is that there might be cases (at least when I run those scripts out of the context of UI5 Tooling meta repo i.e. fork of ui5-tooling poject on my own GH account) where the results of those scripts are null. Then, for example, `schema` could have changed, but the `api` might not be build. Failing any of those copies, would fail the final push to the `gh-pages` branch. And yet, there might be resources that could have been published. This is the core reasoning behind those.
Do you think thare might be a better handling for those cases? Or we could ignore entirelly the possibility not to copy everything?

Reply:
Removed the error supression